zkdb/src/conversion/bson-test.ts

```diff
@@ -0,0 +1,140 @@
+import { serializer, deserializer } from './bson.js';
+import { BSON } from 'bson';
+import {
+    Poseidon,
+    Field,
+    Bool,
+    isReady,
+    Struct,
+    shutdown,
+    Group,
+    PublicKey,
+    UInt32,
+    UInt64,
+    PrivateKey,
+    Signature,
+    Int64,
+    Encoding
+} from 'snarkyjs';
+
+await isReady;
+
+
+console.log("==============================================================")
+console.log("======================Testing SnarkyJS========================")
+
+// Test for serializing SnarkyJS struct
+class RandomData extends Struct({
+    publicKey: PublicKey,
+    uint32: UInt32,
+    uint64: UInt64,
+    field: Field,
+    group: Group,
+    bool: Bool,
+    signature: Signature,
+    int64: Int64
+}) {
+    constructor(
+        publicKey: PublicKey,
+        uint32: UInt32,
+        uint64: UInt64,
+        field: Field,
+        group: Group,
+        bool: Bool,
+        signature: Signature,
+        int64: Int64
+    ) {
+        super({ publicKey, uint32, uint64, field, group, bool, signature, int64 });
+        this.publicKey = publicKey;
+        this.uint32 = uint32;
+        this.uint64 = uint64;
+        this.field = field;
+        this.group = group;
+        this.bool = bool;
+        this.signature = signature;
+        this.int64 = int64;
+    }
+
+    hash(): Field {
+        return Poseidon.hash(this.publicKey.toFields().concat(this.uint32.toFields()).concat(this.field.toFields()));
+    }
+
+    addPoints(n: number): RandomData {
+        return new RandomData(this.publicKey, this.uint32.add(n), this.uint64.add(n), this.field, this.group, this.bool, this.signature, this.int64)
+    }
+}
+
+let randomAccountPk = PrivateKey.random()
+let randAccount = randomAccountPk.toPublicKey()
+
+let msg = Encoding.stringToFields("New signature");
+let signature = Signature.create(randomAccountPk, msg);
+
+let bob = new RandomData(randAccount, UInt32.from(5), UInt64.from(6), Field(25), Group.fromFields([Field(134), Field(3)]), Bool(false), signature, Int64.from(57));
+
+console.log("======================Input SnarkyJS========================")
+
+console.log(bob.publicKey.toBase58());
```

Comment:
We should use `assert()` instead of `console.log()`. We are using `jest` but non of test case follow jest.
https://jestjs.io/docs/getting-started

Reply:
Done.